src/lib/y2storage/proposal/assigned_space.rb

```diff
@@ -56,12 +56,14 @@ def partition_type
         @partition_type if @partition_type_calculated
 
         @partition_type_calculated = true
-        @partition_type = if disk.partition_table.extended_possible
-          if disk.partition_table.has_extended
-            inside_extended? ? :logical : :primary
+        disk.as_not_empty do
```

Comment:
Hm... in the old code, the difference was clearly visible. Now it's pretty mixed up IMHO; it is not immediately obvious that one code branch is only for the "completely empty disk" case.